cachito/workers/pkg_managers/pip.py

```diff
@@ -1575,3 +1578,104 @@ def get_hosted_repositories_username(request_id):
     :rtype: str
     """
     return f"cachito-pip-{request_id}"
+
+
+def generate_pip_conf_content(nexus_pypi_hosted_repo_url, username, password, custom_ca_path=None):
+    """
+    Generate a pip.conf file with the nexus repo url and authentication configured.
+
+    :param str nexus_pypi_hosted_repo_url: the URL for the Nexus PyPI hosted repository
+        for the request
+    :param str username: the username of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str password: the password of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str custom_ca_path: the path to set ``cert`` to in the pip.conf file; if not provided,
+        this option will be omitted
+    :return: the contents of the pip.conf file
+    :rtype: str
+    """
+    # Instead of getting the token from Nexus, use basic authentication as supported by Nexus:
+    # https://pip.pypa.io/en/stable/user_guide/#basic-authentication-credentials
+    # Create the request specific nexus repo url to be passed as `index-url` within pip.conf
+    if "://" in nexus_pypi_hosted_repo_url:
+        nexus_pypi_hosted_repo_url = nexus_pypi_hosted_repo_url.replace(
+            "://", f"://{username}:{password}@", 1
+        )
+    else:
+        raise CachitoError(
+            f"Nexus PyPI Hosted Repo URL: {nexus_pypi_hosted_repo_url} is not a valid URL"
+        )
+
+    pip_conf_content = textwrap.dedent(
+        f"""\
+        [global]
+        index-url = {nexus_pypi_hosted_repo_url}
+        """
+    )
+    if custom_ca_path:
+        # The CA could be embedded in the actual pip.conf file with the `cert` option, but this
+        # would mean that the CA file contents would be duplicated in the Cachito database for
+        # every request that uses a pip conf file
+        pip_conf_content += f'cert = "{custom_ca_path}"\n'
+
+    return pip_conf_content
+
+
+def create_custom_pip_conf_file(request_id, package_configs=None):
```

Comment:
I found the NPM equivalent of this code in `tasks.npm.fetch_sources`. Is this module the right place for this function? Should it be in `tasks.pip` instead?

(or, taking this a step further, is this the right PR to add this functionality?)

Reply:
The fetch sources will be a task for python as well in `tasks.pip`. However, the `generate_npmrc_content` is in `pkg_managers.general_js`.

To have an analogous implementation, generating the file contents should be in pkg_managers and this will be used in a task.

Thoughts?

Reply:
>The fetch sources will be a task for python as well in tasks.pip. However, the generate_npmrc_content is in pkg_managers.general_js.

>To have an analogous implementation, generating the file contents should be in pkg_managers and this will be used in a task.

I agree! By `pkg_managers`, did you mean `pkg_managers/pip.py` ?

>Thoughts?

Reply:
Right, `generate_pip_conf_content` belongs here for sure, what I'm concerned about is the second function, which generates data to put in the Cachito response. I'm not *that* opposed to having it here, fair warning though: if we're keeping it, I will have a couple more comments about the body of the function :D

Reply:
Yes, @chmeliik has a point  here.

This should either go to the tasks.pip module or be left to be implemented in the fetch_sources work as in the npm module.

Reply:
Since we already have it implemented, I might just move it to `tasks/pip.py` over removing it and then it can be used within fetch_sources there.

@chmeliik I'm waiting for your comments on the body of the function :)